Crucible/SchemaCore/SchemaTranslators/ClassTranslator.cs

```diff
@@ -0,0 +1,227 @@
+            outputValue = (TCastType)(object)Traverse(value);
+            return true;
+          }
+        }
+        catch
+        {
+          outputValue = default;
+          return false;
+        }
+      }
+      else if (typeof(IEnumerable).IsAssignableFrom(castType))
+      {
+        if(typeof(IEnumerable).IsAssignableFrom(valueType))
+        {
+          try
+          {
+            MethodInfo getEnumeratorMethod = valueType.GetMethod("GetEnumerator");
+            object enumerator = getEnumeratorMethod.Invoke(value, null);
+            Type enumeratorType = enumerator.GetType();
+            MethodInfo moveNext = enumeratorType.GetMethod("MoveNext");
+            PropertyInfo current = enumeratorType.GetProperty("Current");
+            if (castType.IsGenericType)
+            {
+              Type[] genericArguments = castType.GetGenericArguments();
+              object returnValue = Activator.CreateInstance(castType);
+              MethodInfo add = returnValue.GetType().GetMethod("Add");
+              
+              MethodInfo tryCastType = this.GetType().GetMethod("InnerTryCastValue",BindingFlags.NonPublic | BindingFlags.Instance);
+              MethodInfo castToInnerType = tryCastType.MakeGenericMethod(genericArguments[0]);
+              while ((bool)moveNext.Invoke(enumerator, null))
+              {
+                object currentValue = current.GetValue(enumerator);
+                Type currentValueType = currentValue.GetType();
+                object innerConvertedValue = null;
+                object[] args = new object[] { currentValue, currentValueType, genericArguments[0], innerConvertedValue };
+                castToInnerType.Invoke(this, args);
+                add.Invoke(returnValue, new object[] { args[3] });
+              }
+              outputValue = (TCastType)returnValue;
+              return true;
+            }
+            else
+            {
+              List<object> returnValue = new List<object>();
+              while ((bool)moveNext.Invoke(enumerator, null))
+              {
+                returnValue.Add(current.GetValue(enumerator));
+              }
+              if(castType == typeof(JArray))
+              {
+                outputValue = (TCastType)(object)JArray.FromObject(returnValue);
+                return true;
+              }
+              else
+              {
+                outputValue = (TCastType)(object)returnValue;
+                return true;
+              }
+            }
+          }
+          catch
+          {
+            outputValue = default;
+            return false;
+          }
+        }
+        else
+        {
+          outputValue = default;
+          return false;
+        }
+      }
+      else
+      {
+        try
+        {
+          outputValue = (TCastType)Convert.ChangeType(value, typeof(TCastType));
+          return true;
+        }
+        catch
+        {
+          outputValue = default;
+          return false;
+        }
```

Comment:
Where would we return this exception information? We don't want schema validation to stop because an exception was thrown here. We could return List<Error> instead of bool and raise a new error, but we're already capturing the fact that the cast failed, and that would almost certainly be slower than just returning a bool. Is that worth it?